src/test/java/catan/controllers/testcases/play/BuildCityTest.java

```diff
@@ -0,0 +1,216 @@
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+
+//@SpringApplicationConfiguration(classes = {TestApplicationConfig.class, RequestResponseLogger.class})  // if needed initial request and JSON response logging:
+//@SpringApplicationConfiguration(classes = TestApplicationConfig.class)
+@SpringApplicationConfiguration(classes = TestApplicationConfig.class)
+@WebIntegrationTest("server.port:8091")
+public class BuildCityTest extends PlayTestUtil {
+
+    public static final String USER_NAME_1 = "user1_BuildCityTest";
+    public static final String USER_PASSWORD_1 = "password1";
+    public static final String USER_NAME_2 = "user2_BuildCityTest";
+    public static final String USER_PASSWORD_2 = "password2";
+    public static final String USER_NAME_3 = "user3_BuildCityTest";
+    public static final String USER_PASSWORD_3 = "password3";
+
+    private static boolean initialized = false;
+
+    @Autowired
+    private RandomUtil randomUtil;
+
+    private Scenario scenario;
+
+    @Before
+    public void setup() {
+        if (!initialized) {
+            registerUser(USER_NAME_1, USER_PASSWORD_1);
+            registerUser(USER_NAME_2, USER_PASSWORD_2);
+            registerUser(USER_NAME_3, USER_PASSWORD_3);
+            initialized = true;
+        }
+
+        scenario = new Scenario((RandomUtilMock) randomUtil);
+    }
+
+    @Test
+    public void should_successfully_build_city_even_if_user_does_not_have_resources_in_preparation_stage() {
+        startNewGame(12, 2)
+                .BUILD_SETTLEMENT(1).atNode(2, -2, "topLeft")
+                .BUILD_ROAD(1).atEdge(2, -2, "topLeft")
+                .END_TURN(1)
+
+                .BUILD_SETTLEMENT(2).atNode(2, -1, "bottomRight")
+                .BUILD_ROAD(2).atEdge(2, -1, "bottomRight")
+                .END_TURN(2)
+
+                .BUILD_SETTLEMENT(3).atNode(0, 2, "topRight")
+                .BUILD_ROAD(3).atEdge(0, 2, "topRight")
+                .END_TURN(3)
+
+                .getGameDetails(3).gameUser(3).check("resources.wheat", is(0))
+                .getGameDetails(3).gameUser(3).check("resources.stone", is(0))
+
+                .BUILD_CITY(3).atNode(0, 0, "bottomRight").successfully()
+
+                .getGameDetails(3).gameUser(3).check("resources.wheat", is(0))
+                .getGameDetails(3).gameUser(3).check("resources.stone", is(0));
+    }
+
+    @Test
+    public void should_successfully_build_city_if_user_has_enough_resources_in_main_stage() {
+        startNewGame(12, 1);
+        playPreparationStage();
+        giveResourcesToFirstPlayerForCityBuilding()
+                .nextRandomDiceValues(asList(6, 6))
+                .THROW_DICE(1)
+
+                .getGameDetails(1).gameUser(1).check("resources.wheat", is(2))
+                .getGameDetails(1).gameUser(1).check("resources.stone", is(3))
+
+                .BUILD_CITY(1).atNode(1, -1, "top").successfully()
+
+                .getGameDetails(1).gameUser(1).check("resources.wheat", is(0))
```

Comment:
this check doesn't answer on question in test case, I would create another test case for it, or rename the test case